networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/rpc/RpcMessageHandler.java

```diff
@@ -70,12 +67,15 @@ public RpcMessageHandler(
         .thenCompose(ctr -> ctr.invoke(request));
   }
 
-  private CompletableFuture<TResponse> invokeLocal(Connection connection, TRequest request) {
-    return wrapInFuture(
-        () -> {
-          final Peer peer = peerLookup.getPeer(connection);
-          return localMessageHandler.onIncomingMessage(peer, request);
-        });
+  private void invokeLocal(
+      Connection connection, TRequest request, ResponseCallback<TResponse> callback) {
+    try {
+      final Peer peer = peerLookup.getPeer(connection);
```

Comment:
Do you need to handle the case where `peer == null`?

Reply:
Conveniently with libp2p, no. The Peer is added to `PeerManager` when the connection is first established and only removed when the connection disconnects so we can never get a message while the Peer isn't present in `PeerManager`.